util/plugins/eu.esdihumboldt.util/src/eu/esdihumboldt/util/geometry/interpolation/ArcInterpolation.java

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (c) 2016 wetransform GmbH
+ * 
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution. If not, see <http://www.gnu.org/licenses/>.
+ * 
+ * Contributors:
+ *     wetransform GmbH <http://www.wetransform.to>
+ */
+
+package eu.esdihumboldt.util.geometry.interpolation;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.GeometryFactory;
+import com.vividsolutions.jts.geom.LineString;
+
+import de.fhg.igd.slf4jplus.ALogger;
+import de.fhg.igd.slf4jplus.ALoggerFactory;
+
+/**
+ * Interpolation of a geometry. Specifically for Arc
+ * 
+ * @author Arun
+ */
+public class ArcInterpolation extends Interpolation<LineString> {
+
+	private static final ALogger log = ALoggerFactory.getLogger(ArcInterpolation.class);
+
+	/**
+	 * Constructor
+	 * 
+	 * @param coordinates raw geometry coordinates
+	 * @param maxPositionalError maximum positional error
+	 */
+	public ArcInterpolation(Coordinate[] coordinates, double maxPositionalError) {
+		super(coordinates, maxPositionalError);
+	}
+
+	/**
+	 * @see eu.esdihumboldt.util.geometry.interpolation.Interpolation#validateRawCoordinates()
+	 */
+	@Override
+	protected boolean validateRawCoordinates() {
+		if (rawGeometryCoordinates.length != 3) {
+			log.error("Invalid arc geometry. Arc must be represented by 3 points in GML.");
```

Comment:
Just a note - you decided to use the 3 point representation for the interpolation, that's fine, but the class is not GML specific and the representation neither, so I would remove the GML reference here.